client/src/main/scala/io/delta/sharing/client/DeltaSharingProfileProvider.scala

```diff
@@ -17,22 +17,138 @@
 package io.delta.sharing.client
 
 import java.nio.charset.StandardCharsets.UTF_8
+import java.util.Locale
 
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.{DeserializationContext, JsonDeserializer}
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.node.ObjectNode
 import org.apache.commons.io.IOUtils
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 import org.apache.spark.delta.sharing.TableRefreshResult
 
+import io.delta.sharing.client.DeltaSharingProfile.validateNotNullAndEmpty
 import io.delta.sharing.client.util.JsonUtils
 
-case class DeltaSharingProfile(
-    shareCredentialsVersion: Option[Int],
-    endpoint: String = null,
+@JsonDeserialize(using = classOf[DeltaSharingProfileDeserializer])
+sealed trait DeltaSharingProfile {
+  val shareCredentialsVersion: Option[Int]
+  val endpoint: String
+  val `type` : String
+
+  private [client] def validate(): Unit = {
+    if (shareCredentialsVersion.isEmpty) {
+      throw new IllegalArgumentException(
+        "Cannot find the 'shareCredentialsVersion' field in the profile file")
+    }
+
+    if (shareCredentialsVersion.get > DeltaSharingProfile.CURRENT) {
```

Comment:
perhaps also check that it is >= 1?

Reply:
We already have a check for this in the sublcass please see here:
https://github.com/delta-io/delta-sharing/pull/553/files#diff-41000a1f5b33604158b33b25b259a9892ecc0ed468733bc4a71704e89b91fe56R68

and here https://github.com/delta-io/delta-sharing/pull/553/files#diff-41000a1f5b33604158b33b25b259a9892ecc0ed468733bc4a71704e89b91fe56R86